pkg/plugins/policies/meshratelimit/plugin/v1alpha1/plugin.go

```diff
@@ -115,6 +126,54 @@ func applyToInbounds(
 	return nil
 }
 
+func applyToEgress(rs *core_xds.ResourceSet, proxy *core_xds.Proxy) error {
+	listeners := xds.GatherListeners(rs)
+	for _, resource := range proxy.ZoneEgressProxy.MeshResourcesList {
+		for _, es := range resource.ExternalServices {
+			meshName := resource.Mesh.GetMeta().GetName()
+			esName, ok := es.Spec.GetTags()[mesh_proto.ServiceTag]
+			if !ok {
+				continue
+			}
+			policies, ok := resource.Dynamic[esName]
+			if !ok {
+				continue
+			}
+			mrl, ok := policies[api.MeshRateLimitType]
+			if !ok {
+				continue
+			}
+			if listeners.Egress == nil {
+				log.V(1).Info("skip applying MeshRateLimit, Egress has no listener",
+					"proxyName", proxy.ZoneEgressProxy.ZoneEgressResource.GetMeta().GetName(),
+					"mesh", resource.Mesh.GetMeta().GetName(),
+				)
+				return nil
+			}
```

Comment:
Why are we doing this check here and not immidiately after gathering listeners in line 130?